### PR TITLE
Reduce roadhog ultimate duration

### PR DESCRIPTION
**Pull request by novalderya**
hog_ult

Reduce roadhog ultimate duration, logic based on the cancelling of the ultimate of the status

Issue : Does not take into account the state of Hog before respawning so for example if he is nade by Ana I think it will reset it

Fix issue #224

### CODE REVIEW

File: src/ow1/reset_hero.opy [hunk 1]
```diff
@@ -71,3 +72,14 @@ rule "[reset_hero.opy]: Reset hero on hero switch":
...
     resetHero()
+
+
+def respawnSameState():
+    @Name "[reset_hero.opy]: respawnSameState()"
+    eventPlayer.player_health = eventPlayer.getHealth()
+    kill(eventPlayer,null)
+    # Use resurrect to keep the same position and angle
+    eventPlayer.resurrect()
+    # Wait to set healt else roadhog will be full life
```
Comment (MrKingMichael) [inline src/ow1/reset_hero.opy:83]: Just fix this little typo and we're set.

Comment (snappycreeper) [reply src/ow1/reset_hero.opy:83]: dont use resurrect

**Comment by MrKingMichael** [comment]
Works perfectly. Can't find any bugs in my testing.

File: src/ow1/reset_hero.opy (hunk 1 above)
Comment (novalderya) [reply src/ow1/reset_hero.opy]: Okay what is the reason to not use it ?

**Comment by MrKingMichael** [comment]
It seems that Ana's healing boost works just fine too.

File: src/ow1/reset_hero.opy (hunk 1 above)
Comment (MrKingMichael) [reply src/ow1/reset_hero.opy]: I was wary of that too but it works in my testing.

Comment (snappycreeper) [reply src/ow1/reset_hero.opy]: > I was wary of that too but it works in my testing. Roadhog still takes damage as expected.

Hm ecksdee said you don't take damage until 3 seconds after and I had the same problem. Maybe set health fixes it.

Comment (MaxwellJung) [reply src/ow1/reset_hero.opy]: > dont use resurrect

The resurrected hog player is immune to damage after rez until he uses an ability. @novalderya

**Comment by MaxwellJung** [comment]
@MrKingMichael have you tried shooting the hog after he's rezzed?

I didn't test this PR, but on the other hog cc stun PR #225, when I use wholehog on account 1 then hook account 1 player using account 2 then shoot account 1 player after resurrect, account 1 player was immune to any damage (although account 2 player still got hitmarkers and ult charge).

**Comment by novalderya** [comment]
> @MrKingMichael have you tried shooting the hog after he's rezzed?
> 
> I didn't test this PR, but on the other hog cc stun PR #225, when I use wholehog on account 1 then hook account 1 player using account 2 then shoot account 1 player after resurrect, account 1 player was immune to any damage (although account 2 player still got hitmarkers and ult charge).

I tested with some bots and I lose life after my ult :
hog_ult_dmg

**Comment by MaxwellJung** [comment]
> > @MrKingMichael have you tried shooting the hog after he's rezzed?
> > I didn't test this PR, but on the other hog cc stun PR #225, when I use wholehog on account 1 then hook account 1 player using account 2 then shoot account 1 player after resurrect, account 1 player was immune to any damage (although account 2 player still got hitmarkers and ult charge).
> 
> I tested with some bots and I lose life after my ult : hog_ult_dmg

Ok if that's the case, I"ll merge to staging build then test before merging to main branch.

**Review by MaxwellJung — APPROVED** [review]
Merging to staging

**Pull request merged**